frontend-react/src/pages/HomePage.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect, useCallback } from "react"
+import { useApiClient } from "@/hooks/useApiClient"
+import SearchForm from "@/components/HomePage/SearchForm"
+import { ApiErrorResponse, SelectOption } from "@/types"
+import { AxiosError } from "axios"
+
+export default function HomePage() {
+  const [sportsTypes, setSportsTypes] = useState<SelectOption[]>([])
+  const [sportsTypeSelected, setSportsTypeSelected] = useState<SelectOption | null>(null)
+  const [errors, setErrors] = useState<string[]>([])
+  const apiClient = useApiClient()
+
+  const fetchSportsTypes = useCallback(async () => {
+    try {
+      setErrors([])
+      const res = await apiClient.get("/sports_types")
+      setSportsTypes(res.data.data)
+    } catch (error: unknown) {
+      const axiosError = error as AxiosError<ApiErrorResponse>
+      setErrors([axiosError.response?.data?.error || "競技を表示できませんでした。"])
+    }
+  }, [apiClient])
+
+  useEffect(() => {
+    fetchSportsTypes()
+  }, [fetchSportsTypes])
```

Comment:
これだと不都合あったりしますかね？？(問題なければこれの方がわかりやすそう)
```suggestion
 useEffect(() => {
    const fetchSportsTypes = async () => {
      try {
        setError(null)
        const res = await apiClient.get("/sports_types")
        setSportsTypes(res.data.data)
      } catch (err) {
        handleError(err)
      }
    }

    fetchSportsTypes()
  }, [apiClient])
```